Added a range of numerical values for testing the model

made-line opened the pull request:
I added in a range of values for numerical features so users know the scale when testing a model. Also updated the spacing in between continuous features. 

Before: 
![Screen Shot 2021-03-22 at 4 41 00 PM](https://user-images.githubusercontent.com/40412372/112071878-ed8a8700-8b2d-11eb-93af-3e672188b750.png)


<img width="278" alt="Screen Shot 2021-03-23 at 4 18 04 PM" src="https://user-images.githubusercontent.com/40412372/112231303-b54d7c00-8bf3-11eb-8eb3-19b493124a84.png">

After: 
![Screen Shot 2021-03-29 at 4 49 10 PM](https://user-images.githubusercontent.com/40412372/112913546-b298ce00-90ae-11eb-9277-60ac9f4bb5a3.png)

Changes to spacing in the context of Code Studio
Before:
![Screen Shot 2021-03-29 at 4 25 17 PM](https://user-images.githubusercontent.com/40412372/112915157-972fc200-90b2-11eb-9b73-0ea4a72f7ec3.png)

After: 
![Screen Shot 2021-03-29 at 5 18 34 PM](https://user-images.githubusercontent.com/40412372/112915292-d5c57c80-90b2-11eb-94ae-919267145d84.png)

breville commented:
Do students understand this nomenclature? 

We show range in the popup for a numerical column but we just show "min", "max", and "range".  Should we be consistent here?

Finally, should we truncate to maybe 1 or 2 decimal places to make it more readable?

Erin007 commented:
> We show range in the popup for a numerical column but we just show "min", "max", and "range". Should we be consistent here?

+1 to making it more clear what these numbers are. Maybe something like "Age (min: 1, max: 84)". Currently this looks like an array, which might be confusing to students.  I don't think we need range in this scenario.

made-line commented:
Thank you both! I made changes accordingly and updated the "After" screenshot :)

breville commented on src/UIComponents/Predict.jsx:
```diff
@@ -47,7 +49,10 @@ class Predict extends Component {
                   <div key={index}>
                     <label>
-                      {feature}: &nbsp;
+                      {feature} (min:{" "}
+                      {+this.props.rangesByColumn[feature].min.toFixed(2)}, max:{" "}
+                      {+this.props.rangesByColumn[feature].max.toFixed(2)}
```
Could we hoist these long expressions into a couple local variables in the `render` function, and then use string interpolation to make this expression more compact?  

Also are the `+`s necessary?

made-line replied:
Good point! If I put the variables in the render function, it loses the context of feature from selectedNumericalFeatures.map((feature, index). Do we want to do this or is the change I just pushed okay?

breville replied:
Oh I see.  There are a few ways to factor this, such as having another React component, or a sub-render function, or a new data structure entirely (such as we do with `getCrossTabData`) but I think this is okay as-is.

made-line replied:
Those are good things for me to think about going forward. Thank you!

breville commented:
Just curious about the screenshots: why are the first two items touching, while the third is separate?  While we are in here, could we tidy up that spacing, especially between the first two?

made-line commented:
Nice catch. I just added the same padding that's applied to categorical
features. I also added a new screenshot to the PR. Let me know if you think
the padding should be adjusted.



On Sat, Mar 27, 2021 at 9:57 PM Brendan Reville ***@***.***>
wrote:

> Just curious about the screenshots: why are the first two items touching,
> while the third is separate? While we are in here, could we tidy up that
> spacing, especially between the first two?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/code-dot-org/ml-playground/pull/122#issuecomment-808844622>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJUKJVFQ7ZS2B5YRFWS7VYDTF2ZLVANCNFSM4ZUC4AWQ>
> .
>
